atc/engine/builder/step_factory.go

```diff
@@ -14,15 +14,16 @@ import (
 )
 
 type stepFactory struct {
-	pool                  worker.Pool
-	client                worker.Client
-	resourceFactory       resource.ResourceFactory
-	teamFactory           db.TeamFactory
-	resourceCacheFactory  db.ResourceCacheFactory
-	resourceConfigFactory db.ResourceConfigFactory
-	defaultLimits         atc.ContainerLimits
-	strategy              worker.ContainerPlacementStrategy
-	lockFactory           lock.LockFactory
+	pool                           worker.Pool
+	client                         worker.Client
+	resourceFactory                resource.ResourceFactory
+	teamFactory                    db.TeamFactory
+	resourceCacheFactory           db.ResourceCacheFactory
+	resourceConfigFactory          db.ResourceConfigFactory
+	defaultLimits                  atc.ContainerLimits
+	strategy                       worker.ContainerPlacementStrategy
+	lockFactory                    lock.LockFactory
+	enableRerunWhenWorkerDisappear bool
```

Comment:
Small critique - again would add an "s" at the end `enableRerunWhenWorkerDisappears`

Reply:
fixed